File: samples/CslaNet/vb/DeepData/DeepData.DAL.Direct/OrderData.vb
Imports System.Configuration.ConfigurationManager
Imports System.Data.SqlClient
Imports Csla.Data

Public Class OrderData
  Inherits DeepData.DAL.OrderData

  Private _cn As SqlConnection
  Private _cm As SqlCommand
  Private _data As SafeDataReader

  Public Sub New()

    _cn = New SqlConnection(ConnectionStrings("DeepData.My.MySettings.DeepDataStoreConnectionString").ConnectionString)
    _cn.Open()
    _cm = _cn.CreateCommand
    _cm.CommandText = "SELECT id,customer FROM [Order];SELECT orderid,id,product FROM OrderLine;SELECT orderid,lineid,id,detail FROM OrderLineDetail"
    _cm.CommandType = CommandType.Text
    _data = New SafeDataReader(_cm.ExecuteReader)

  End Sub

  Public Overrides Function GetOrders() As Object

    Return _data

  End Function

#Region " IDisposable Support "

  ' IDisposable
  Protected Overrides Sub Dispose(ByVal disposing As Boolean)
    If disposing Then
      _data.Dispose()
      _cm.Dispose()
      _cn.Dispose()
    End If
  End Sub

#End Region

End Class

File: Samples/CslaNet/cs/DeepData/DeepData.DAL.Direct/OrderData.cs
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using Csla.Data;

namespace DeepData.DAL.Direct
{
	///<summary>
	///</summary>
	public class OrderData
	  : DeepData.DAL.OrderData
	{
		private SqlConnection _cn;
		private SqlCommand _cm;
		private SafeDataReader _data;

		///<summary>
		///</summary>
		public OrderData()
		{
			_cn = new SqlConnection(ConfigurationManager.ConnectionStrings["DeepDataDAL"].ConnectionString);
			_cn.Open();
			_cm = _cn.CreateCommand();
			_cm.CommandText =
				"SELECT id,customer FROM [Order];SELECT orderid,id,product FROM OrderLine;SELECT orderid,lineid,id,detail FROM OrderLineDetail";
			_cm.CommandType = CommandType.Text;
			_data = new SafeDataReader(_cm.ExecuteReader());
		}

		///<summary>
		///</summary>
		///<returns></returns>
		public override object GetOrders()
		{
			return _data;
		}


		#region " IDisposable Support "

		// IDisposable
		///<summary>
		///Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
		///</summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				_data.Dispose();
				_cm.Dispose();
				_cn.Dispose();
			}
		}

		#endregion

	}
}